### ISSUE TITLE
Plan Route, edit track: user interface adaptation of track update functionality

### ISSUE THREAD

**Issue by Gbernardy** [osmandapp/OsmAnd#13839]
# 🚀 feature request

### Describe the solution you'd like

I have tried several times the trace update features ... in my opinion they need a slight adjustment to make them more homogeneous.

Below I describe what adjustments you should make:

On-screen command "done"
------------------------------------------

The command "done" on the screen, if executed, saves the track and interrupts the edit session of the same.  It would be more useful to have the "save" command on the screen which saves and does not interrupt the edit session of the track.  To go back to the previous screen "myplaces" the Android key "go back" can be used.

Commands on the options menu panel
------------------------------------------

![IMG_20220303_083145](https://user-images.githubusercontent.com/99968793/156520289-1bad6fac-cc03-4d7e-8b89-21a77545fc28.jpg)

Command "save changes and exit"

The command should save the edited track go back to "my places" screen (as now for "save changes").

Command "save as new track"

The command should save the edited track with the new name and CONTINUE the edit session of new track.

Command "add to a track"

The command should save the union of the two tracks in a new track and CONTINUE the edit session (as save in new track)(now save over the second track without warning).

The following panel (used only in "add to a track") would be deleted.  The "share" functionality, if required in edit ... could be added to the options panel)

![Screenshot_2022-03-03-08-56-59-53_54166e5c6f8ad9491c65e2397c495c92](https://user-images.githubusercontent.com/99968793/156522522-0919c360-43ca-48ee-9a14-040afe341009.jpg)

**Comment by Gbernardy** [comment]
Plan Route, incorrect go back management.
---

Open track from "my place" screen

![Screenshot_2022-03-04-13-04-28-62_54166e5c6f8ad9491c65e2397c495c92](https://user-images.githubusercontent.com/99968793/156761418-b1ccdf22-9d5a-4c25-b285-63e609f631d4.jpg)

Short tap on track edit

![IMG_20220304_130546](https://user-images.githubusercontent.com/99968793/156761462-f0cacea5-cb60-42a0-86e0-4e99990329f1.jpg)

Open edit session on screen,
and immediately after ....
short tap on go back android

![IMG_20220304_130732](https://user-images.githubusercontent.com/99968793/156761527-d0b7b5c5-123c-41e2-ac69-d85ebfa5bdc8.jpg)

return to "My place" screen
and immediately after ...
short tap on go back android

![IMG_20220304_130914](https://user-images.githubusercontent.com/99968793/156761624-691c994c-7fe4-4a58-bcae-14bb92b91551.jpg)

return to edit session of track ...

![Screenshot_2022-03-04-13-09-52-11_54166e5c6f8ad9491c65e2397c495c92](https://user-images.githubusercontent.com/99968793/156761680-228a9e9b-f45f-4b12-ad2b-c38b0bd9772b.jpg)

The edit session remained open ... this is a bug.
----

**Comment by Gbernardy** [comment]
The incorrect management of the edit sessions of the traces is the only cause of the anomalous management of the move point function

![IMG_20220306_162526](https://user-images.githubusercontent.com/99968793/156929929-ba3fe389-8f8b-400a-9fcd-4a5440a66617.jpg)

ADD button enabled and not button APPLY !!! 
This is a another bug.
-----------

**Comment by Gbernardy** [comment]
The bug is more evident if instead of reopening the same track ...
i open another one ...
------

**Comment by Lee-Carre** [comment]
Please stop abusing markup (headings, in this case).

**Comment by Gbernardy** [comment]
> Please stop abusing markup (headings, in this case).

It is the "-" character that changes the size of the previous characters ... I was not going to achieve this.

![IMG_20220307_123446](https://user-images.githubusercontent.com/99968793/157025515-bd45caf0-bbca-4e9e-94c3-f13d16ac3168.jpg)

**Comment by vshcherb** [comment]
It's hard to follow this conversations, screenshots are too big and lots of headers / bold text. Consider to updated screenshots by using html img tags like 
```<img src ="... url ... " width="20%" height="20%" /> ```

Please keep the conversation short and precise .

**Comment by Gbernardy** [comment]
The editing sessions of the tracks, not always managed correctly by OsmAnd, sometimes give rise to this anomaly ... hamburger button disappeared from the screen.

![IMG_20220308_181652](https://user-images.githubusercontent.com/99968793/157433173-7af3c961-6262-4d80-9656-880754abdaad.jpg)
